dev/tests/hu.elte.txtuml.validation.test/src/hu/elte/txtuml/validation/SequenceDiagramTest.java

```diff
@@ -0,0 +1,82 @@
+package hu.elte.txtuml.validation;
+
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Mockito.verify;
+import static hu.elte.txtuml.validation.sequencediagram.ValidationErrors.SEND_EXPECTED;
+import static hu.elte.txtuml.validation.sequencediagram.ValidationErrors.INVALID_LIFELINE_DECLARATION;
+import static hu.elte.txtuml.validation.sequencediagram.ValidationErrors.INVALID_POSITION;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.hamcrest.Description;
+import org.junit.Test;
+import org.mockito.ArgumentMatcher;
+
+import hu.elte.txtuml.validation.common.ValidationProblem;
+import hu.elte.txtuml.validation.sequencediagram.ValidationErrors;
+import hu.elte.txtuml.validation.sequencediagram.visitors.SequenceDiagramVisitor;
+
+public class SequenceDiagramTest extends ValidationTestBase {
+
+	private static final String VALIDATION_EXAMPLES_PACKAGE = "/hu/elte/txtuml/examples/validation/sequencediagram/";
+
+	@Override
+	protected String getValidationExamplesPackage() {
+		return VALIDATION_EXAMPLES_PACKAGE;
+	}
+
+	@Test
+	public void testEmptyRunMethod() throws Exception {
+		assertErrors("EmptyRun.java", SEND_EXPECTED);
+	}
+
+	@Test
+	public void testMissingSend() throws Exception {
+		assertErrors("MissingSend.java", SEND_EXPECTED);
+	}
+
+	@Test
+	public void testInvalidLifeline() throws Exception {
+		assertErrors("InvalidLifelineDeclaration.java", INVALID_LIFELINE_DECLARATION, SEND_EXPECTED);
+	}
+
+	@Test
+	public void testInvalidPosition() throws Exception {
+		assertErrors("InvalidPosition.java", INVALID_POSITION, SEND_EXPECTED);
+	}
+
+	@Test
+	public void testValidDiagram() throws Exception {
+		assertValid("ValidDiagram.java");
+	}
+
+	private void assertValid(String fileName) throws IOException {
+		assertErrors(fileName);
+	}
+
+	private void assertErrors(String fileName, ValidationErrors... errorTypes) throws IOException {
+		CompilationUnit compilationUnit = prepareAST(fileName);
+		compilationUnit.accept(new SequenceDiagramVisitor(mockCollector));
+		Arrays.asList(errorTypes).forEach(errorType -> verify(mockCollector).report(is(errorType)));
+		checkNoOtherErrorRaised();
+	}
+
+	private static ValidationProblem is(ValidationErrors type) {
```

Comment:
This method is basically a duplication of the one defined for `ModelErrors`. I believe we could eliminate this using appropriate abstractions.